common/djangoapps/student/views.py

```diff
@@ -1502,6 +1501,9 @@ def create_account_with_params(request, params):
                 request.social_strategy.clean_partial_pipeline()
                 raise ValidationError({'access_token': [error_message]})
 
+    # Perform operations that are non-critical parts of account creation
+    preferences_api.set_user_preference(user, LANGUAGE_KEY, get_language())
```

Comment:
Note to reviewers: this call needs to be made outside of the transaction block above since it has its own `@transaction.commit_on_success` decorator.  So it doesn't rollback the outer transaction if the inner transaction succeeds.  This was an issue that was introduced by the original call to `set_user_preference`, but there was no unit test at the time to verify the transaction rollback.

With django 1.6, we'll be able to use `auto_commit`, which makes nested transactions much easier to deal with.